applicationLibrary/molecularDynamics/tests/singleSiteTests/ATMFunctorHWYTest.cpp

```diff
@@ -0,0 +1,617 @@
+    } else {
+      return mdLib::AxilrodTellerMutoFunctor<Molecule, false, autopas::FunctorN3Modes::Both, true>(_cutoff);
+    }
+  }();
+
+  auto atmFunctorHWY = [&]() {
+    if constexpr (mixing) {
+      return mdLib::AxilrodTellerMutoFunctorHWY<Molecule, true, autopas::FunctorN3Modes::Both, true>(_cutoff, PPL);
+    } else {
+      return mdLib::AxilrodTellerMutoFunctorHWY<Molecule, false, autopas::FunctorN3Modes::Both, true>(_cutoff);
+    }
+  }();
+
+  if constexpr (not mixing) {
+    atmFunctorNoHWY.setParticleProperties(_nu);
+    atmFunctorHWY.setParticleProperties(_nu);
+  }
+
+  ASSERT_TRUE(AoSParticlesEqual(cellHWY, cellNoHWY)) << "Cells not equal after copy initialization.";
+
+  atmFunctorHWY.initTraversal();
+  atmFunctorNoHWY.initTraversal();
+
+  atmFunctorNoHWY.SoALoader(cellNoHWY, cellNoHWY._particleSoABuffer, 0, /*skipSoAResize*/ false);
+  atmFunctorHWY.SoALoader(cellHWY, cellHWY._particleSoABuffer, 0, /*skipSoAResize*/ false);
+
+  ASSERT_TRUE(SoAParticlesEqual(cellHWY._particleSoABuffer, cellNoHWY._particleSoABuffer))
+      << "Cells not equal after loading.";
+
+  if (useUnalignedViews) {
+    atmFunctorNoHWY.SoAFunctorSingle(cellNoHWY._particleSoABuffer.constructView(1, cellNoHWY.size()), newton3);
+    atmFunctorHWY.SoAFunctorSingle(cellHWY._particleSoABuffer.constructView(1, cellHWY.size()), newton3);
+  } else {
+    atmFunctorNoHWY.SoAFunctorSingle(cellNoHWY._particleSoABuffer, newton3);
+    atmFunctorHWY.SoAFunctorSingle(cellHWY._particleSoABuffer, newton3);
+  }
+  ASSERT_TRUE(SoAParticlesEqual(cellHWY._particleSoABuffer, cellNoHWY._particleSoABuffer))
+      << "Cells not equal after applying functor.";
+
+  atmFunctorHWY.SoAExtractor(cellHWY, cellHWY._particleSoABuffer, 0);
+  atmFunctorHWY.SoAExtractor(cellNoHWY, cellNoHWY._particleSoABuffer, 0);
+
+  ASSERT_TRUE(AoSParticlesEqual(cellHWY, cellNoHWY)) << "Cells 1 not equal after extracting.";
+
+  atmFunctorHWY.endTraversal(newton3);
+  atmFunctorNoHWY.endTraversal(newton3);
+
+  double tolerance = 1e-8;
+  EXPECT_NEAR(atmFunctorHWY.getPotentialEnergy(), atmFunctorNoHWY.getPotentialEnergy(), tolerance) << "global uPot";
+  EXPECT_NEAR(atmFunctorHWY.getVirial(), atmFunctorNoHWY.getVirial(), tolerance) << "global virial";
+}
+
+template <bool mixing>
+void ATMFunctorHWYTest::testATMFunctorVSATMFunctorHWYAoS(bool newton3, bool doDeleteSomeParticles) {
+  FMCell cellHWY;
+
+  constexpr size_t numParticles = 7;
+
+  ParticlePropertiesLibrary<double, size_t> PPL{_cutoff};
+  if constexpr (mixing) {
+    PPL.addSiteType(0, 1.);
+    PPL.addATMParametersToSite(0, 1.);
+    PPL.addSiteType(1, 1.5);
+    PPL.addATMParametersToSite(1, 1.);
+    PPL.addSiteType(2, 2.);
+    PPL.addATMParametersToSite(2, 1.);
+    PPL.addSiteType(3, 2.5);
+    PPL.addATMParametersToSite(3, 1.);
+    PPL.addSiteType(4, 3.);
+    PPL.addATMParametersToSite(4, 1.);
+    PPL.calculateMixingCoefficients();
```

Comment:
Add some different `nu` parameters

Reply:
Similar for the rest of the functions (or simply store the PPL as a class member)